### DISCUSSION TITLE
Help following autobuilder.cpp code

### DISCUSSION THREAD

**Discussion by WarrenReynolds** [mjansen4857/pathplanner#396] [General]
In Autobuilder.cpp there are the following lines not inside any class definition
```
bool AutoBuilder::m_configured = false;
std::function<frc2::CommandPtr(std::shared_ptr<PathPlannerPath>)> AutoBuilder::m_pathFollowingCommandBuilder;
std::function<frc::Pose2d()> AutoBuilder::m_getPose;
std::function<void(frc::Pose2d)> AutoBuilder::m_resetPose;
```
How can you just reference the member variable of class without an instance of the class?
Is the autobuilder now a singleton class or something?  If so is there supposed to be some private constructor that maintains a single instance of the class?

**Comment by mjansen4857** [comment]
Everything in auto builder is static now. Those lines initialize the static variables.

**Comment by WarrenReynolds** [comment]
Makes sense.  Showing my lack of understanding of C++ here.  I knew a static variable was shared amongst instances of a class, I just didn't realise that you could access it without going through an instance of the class.  You learn something new every day.
